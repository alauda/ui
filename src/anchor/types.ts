import { EventEmitter, TemplateRef } from '@angular/core';

export interface AnchorItemBase {
  readonly id?: string;
  readonly target?: HTMLElement;
  readonly label: TemplateRef<unknown> | string;
  readonly labelContext?: unknown;
  readonly labelChange?: EventEmitter<TemplateRef<unknown> | string>;
}

export interface AnchorItem extends AnchorItemBase {
  readonly level: number;
}

export interface AnchorTreeItem extends AnchorItemBase {
  children?: AnchorTreeItem[];
}
