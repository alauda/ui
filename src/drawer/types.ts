import { ComponentType } from '@angular/cdk/portal';
import { TemplateRef } from '@angular/core';

import { ValueOf } from '../types';

export const DrawerSize = {
  Small: 'small',
  Medium: 'medium',
  Big: 'big',
} as const;

export type DrawerSize = ValueOf<typeof DrawerSize>;

export interface DrawerOptions<T = unknown, C extends object = object> {
  title?: string | TemplateRef<C>;
  content?: ComponentType<T> | TemplateRef<C>;
  footer?: string | TemplateRef<C>;
  contentParams?: C; // 不仅作为content的参数，同时是title和footer的上下文
  width?: number;
  size?: DrawerSize; // 内置的宽度尺寸，也可以使用 width 自定义
  offsetY?: string;
  divider?: boolean;
  drawerClass?: string;
  visible?: boolean;
  showClose?: boolean;
  mask?: boolean;
  maskClosable?: boolean; // 点击背景是否关闭抽屉
  hideOnClickOutside?: boolean; // 在抽屉外点击是否关闭抽屉，与 maskClosable 的区别是是否有 mask
}
