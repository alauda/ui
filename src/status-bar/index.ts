export * from './status-bar.component';
export * from './status-bar.module';
export * from './status-bar.types';
