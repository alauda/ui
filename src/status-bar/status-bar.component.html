<div [class]="bem.block(size)">
  <ng-container *ngFor="let item of status">
    <div
      *ngIf="item.scale"
      [ngClass]="[bem.element('chunk', item.type), item.class || '']"
      [style.width]="item.scale * 100 + '%'"
      (click)="statusClick.emit(item)"
      [auiTooltip]="template"
      [auiTooltipContext]="item"
    ></div>
  </ng-container>
</div>
