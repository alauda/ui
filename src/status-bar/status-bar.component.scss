@import '../theme/var';
@import '../theme/mixin';

$block: aui-status-bar;
$bar-border-radius: 5px;
$small-bar-border-radius: 4px;

$bar-type: (
  primary: (
    background: use-rgb(primary),
    hover-background: use-rgb(p-2),
  ),
  success: (
    background: use-rgb(green),
    hover-background: use-rgb(g-2),
  ),
  warning: (
    background: use-rgb(yellow),
    hover-background: use-rgb(y-2),
  ),
  error: (
    background: use-rgb(red),
    hover-background: use-rgb(r-2),
  ),
  info: (
    background: use-rgb(n-4),
    hover-background: use-rgb(n-6),
  ),
);

.#{$block} {
  display: inline-flex;
  overflow: hidden;
  width: 100%;
  vertical-align: middle;
  border-radius: $bar-border-radius;

  &--medium {
    height: 10px;
  }

  &--small {
    height: 8px;
    border-radius: $small-bar-border-radius;
    .#{$block}__chunk {
      &:first-child {
        border-top-left-radius: $small-bar-border-radius;
        border-bottom-left-radius: $small-bar-border-radius;
      }

      &:last-child {
        border-top-right-radius: $small-bar-border-radius;
        border-bottom-right-radius: $small-bar-border-radius;
      }
    }
  }

  &__chunk {
    min-width: 2px;
    height: 100%;
    flex: 1 1 1;
    border-radius: use-var(border-radius-m);

    &:first-child {
      border-top-left-radius: $bar-border-radius;
      border-bottom-left-radius: $bar-border-radius;
    }

    &:last-child {
      border-top-right-radius: $bar-border-radius;
      border-bottom-right-radius: $bar-border-radius;
    }

    @each $type in primary, success, warning, error, info {
      $map: map-get($bar-type, $type);
      &--#{$type} {
        background-color: map-get($map, background);
        &:hover {
          background-color: map-get($map, hover-background);
        }
      }
    }

    &--pending {
      height: use-var(inline-height-m);
      animation: pending 0.5s linear infinite;
      background: repeating-linear-gradient(
        -60deg,
        use-rgb(b-5),
        use-rgb(b-5) use-var(spacing-s),
        use-rgb(b-6) use-var(spacing-s),
        use-rgb(b-6) use-var(spacing-m)
      );
    }
  }

  &__chunk + &__chunk {
    margin-left: use-var(spacing-xs);
  }
}

@keyframes pending {
  0% {
    transform: translateY(-50%);
  }
  100% {
    transform: translateY(0);
  }
}
