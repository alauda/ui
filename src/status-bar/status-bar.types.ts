import { ValueOf } from '../internal/types';

export interface Status {
  type?: StatusType;
  class?: string;
  scale: number;
  [key: string]: unknown;
}

export const StatusType = {
  Primary: 'primary',
  Success: 'success',
  Warning: 'warning',
  Error: 'error',
  Info: 'info',
  Pending: 'pending',
} as const;

export type StatusType = ValueOf<typeof StatusType>;

export const StatusBarSize = {
  Medium: 'medium',
  Small: 'small',
} as const;

export type StatusBarSize = ValueOf<typeof StatusBarSize>;
