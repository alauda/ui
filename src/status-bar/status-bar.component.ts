import { NgFor, NgIf, NgClass } from '@angular/common';
import {
  ChangeDetectionStrategy,
  Component,
  EventEmitter,
  Input,
  Output,
  TemplateRef,
  ViewEncapsulation,
} from '@angular/core';

import { TooltipDirective } from '../tooltip/tooltip.directive';
import { Bem, buildBem } from '../utils';

import { Status, StatusBarSize } from './status-bar.types';

@Component({
  selector: 'aui-status-bar',
  templateUrl: './status-bar.component.html',
  styleUrls: ['./status-bar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  preserveWhitespaces: false,
  standalone: true,
  imports: [NgFor, NgIf, NgClass, TooltipDirective],
})
export class StatusBarComponent {
  bem: Bem = buildBem('aui-status-bar');

  @Input()
  status: Status[] = [];

  @Input()
  size: StatusBarSize = StatusBarSize.Medium;

  @Input()
  template: TemplateRef<any>;

  @Output()
  statusClick = new EventEmitter<Status>();
}
