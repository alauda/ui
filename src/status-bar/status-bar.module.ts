import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { TooltipModule } from '../tooltip';

import { StatusBarComponent } from './status-bar.component';

@NgModule({
  imports: [CommonModule, TooltipModule],
  declarations: [StatusBarComponent],
  exports: [StatusBarComponent],
})
export class StatusBarModule {}
