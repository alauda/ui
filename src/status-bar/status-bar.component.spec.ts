import { OverlayContainer } from '@angular/cdk/overlay';
import { Component } from '@angular/core';
import {
  ComponentFixture,
  TestBed,
  fakeAsync,
  inject,
  tick,
} from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { StatusBarModule } from './status-bar.module';
import { Status, StatusType } from './status-bar.types';

describe('Status Bar', () => {
  let fixture: ComponentFixture<TestComponent>;
  let ocEl: HTMLElement;
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [StatusBarModule],
      declarations: [TestComponent],
    });

    fixture = TestBed.createComponent(TestComponent);
    fixture.detectChanges();

    inject([OverlayContainer], (overlayContainer: OverlayContainer) => {
      ocEl = overlayContainer.getContainerElement();
    })();
  });

  it('should match snapshot', fakeAsync(() => {
    expect(fixture).toMatchSnapshot();

    const chunk = fixture.debugElement.query(
      By.css('aui-status-bar .aui-status-bar__chunk'),
    ).nativeElement as HTMLElement;
    chunk.dispatchEvent(new Event('mouseenter'));
    tick(50);

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
    expect(ocEl).toMatchSnapshot();
  }));

  it('should onChunkClick called once', () => {
    const mockFn = jest.fn();

    fixture.componentInstance.onChunkClick = mockFn;

    const chunk = fixture.debugElement.query(
      By.css('aui-status-bar .aui-status-bar__chunk'),
    ).nativeElement as HTMLElement;
    chunk.dispatchEvent(new Event('click'));

    expect(mockFn.mock.calls.length).toBe(1);
    expect(mockFn.mock.calls[0][0]).toEqual({
      scale: 0.1,
      type: StatusType.Info,
      class: 'custom-class',
    });
  });
});

@Component({
  template: `
    <aui-status-bar
      [status]="status1"
      [template]="ref"
      size="medium"
      (statusClick)="onChunkClick($event)"
    ></aui-status-bar>
    <aui-status-bar
      [status]="status1"
      size="small"
      (statusClick)="onChunkClick($event)"
    ></aui-status-bar>
    <aui-status-bar
      [status]="status2"
      size="medium"
      (statusClick)="onChunkClick($event)"
    ></aui-status-bar>
    <aui-status-bar
      [status]="status2"
      size="small"
      (statusClick)="onChunkClick($event)"
    ></aui-status-bar>
    <ng-template
      #ref
      let-type="type"
      let-value="scale"
    >
      <div>{{ type }}: {{ value }}</div>
    </ng-template>
  `,
})
class TestComponent {
  status1: Status[] = [
    {
      scale: 0.1,
      type: StatusType.Info,
      class: 'custom-class',
    },
    {
      scale: 0.3,
      type: StatusType.Error,
    },
    {
      scale: 0.2,
      type: StatusType.Warning,
    },
    {
      scale: 0.2,
      type: StatusType.Success,
    },
    {
      scale: 0.2,
      type: StatusType.Primary,
    },
  ];

  status2: Status[] = [
    {
      type: StatusType.Pending,
      scale: 1,
    },
  ];

  onChunkClick: (status: Status) => {
    //
  };
}
