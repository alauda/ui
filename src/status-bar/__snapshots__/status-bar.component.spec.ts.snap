// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Bar should match snapshot 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  status1={[Function Array]}
  status2={[Function Array]}
>
  <aui-status-bar
    size="medium"
  >
    <div
      class="aui-status-bar aui-status-bar--medium"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--info custom-class"
        style="width: 10%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--error"
        style="width: 30%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--warning"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--success"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--primary"
        style="width: 20%;"
      />
    </div>
  </aui-status-bar><aui-status-bar
    size="small"
  >
    <div
      class="aui-status-bar aui-status-bar--small"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--info custom-class"
        style="width: 10%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--error"
        style="width: 30%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--warning"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--success"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--primary"
        style="width: 20%;"
      />
    </div>
  </aui-status-bar><aui-status-bar
    size="medium"
  >
    <div
      class="aui-status-bar aui-status-bar--medium"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--pending"
        style="width: 100%;"
      />
    </div>
  </aui-status-bar><aui-status-bar
    size="small"
  >
    <div
      class="aui-status-bar aui-status-bar--small"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--pending"
        style="width: 100%;"
      />
    </div>
  </aui-status-bar>
</ng-component>
`;

exports[`Status Bar should match snapshot 2`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  status1={[Function Array]}
  status2={[Function Array]}
>
  <aui-status-bar
    size="medium"
  >
    <div
      class="aui-status-bar aui-status-bar--medium"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--info custom-class"
        style="width: 10%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--error"
        style="width: 30%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--warning"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--success"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--primary"
        style="width: 20%;"
      />
    </div>
  </aui-status-bar><aui-status-bar
    size="small"
  >
    <div
      class="aui-status-bar aui-status-bar--small"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--info custom-class"
        style="width: 10%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--error"
        style="width: 30%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--warning"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--success"
        style="width: 20%;"
      />
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--primary"
        style="width: 20%;"
      />
    </div>
  </aui-status-bar><aui-status-bar
    size="medium"
  >
    <div
      class="aui-status-bar aui-status-bar--medium"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--pending"
        style="width: 100%;"
      />
    </div>
  </aui-status-bar><aui-status-bar
    size="small"
  >
    <div
      class="aui-status-bar aui-status-bar--small"
    >
      <div
        class="aui-status-bar__chunk aui-status-bar__chunk--pending"
        style="width: 100%;"
      />
    </div>
  </aui-status-bar>
</ng-component>
`;

exports[`Status Bar should match snapshot 3`] = `
<div
  class="cdk-overlay-container"
  platform="test"
>
  <div
    class="cdk-overlay-connected-position-bounding-box"
    dir="ltr"
    style="left: 0px; bottom: 0px; height: 0px; width: 0px; align-items: center; justify-content: flex-end;"
  >
    <div
      class="cdk-overlay-pane"
      style="position: static;"
    >
      <aui-tooltip>
        <div
          class="aui-tooltip aui-tooltip--default aui-tooltip--top"
        >
            
          <div>
            info: 0.1
          </div>
          
          
        </div>
      </aui-tooltip>
    </div>
  </div>
</div>
`;
